[Woo POS][Barcodes] Track when a barcode scanning menu item is tapped

joshheald opened the pull request:
<!-- Remember about a good descriptive title. -->

<!-- Id number of the Linear issue this PR addresses. -->
<!-- Part of: # -> use this if your PR is one of many related to one issue -->

## Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

This PR adds tracking for the barcode scanning menu item, so we can determine interest levels in barcode scanning.

## Steps to reproduce
<!-- Step-by-step testing instructions. For new user flows, consider instead stating the goal of the workflow and see if your PR reviewer can accomplish the workflow without specific steps! -->

1. Open POS
2. Tap the `...` menu
3. Tap `Barcode scanning`
4. Observe that the `pos_barcode_scanning_menu_item_tapped` event is tracked.

## Screenshots
<!-- Include before and after images or gifs when appropriate. -->
![CleanShot 2025-06-26 at 11 01 18@2x](https://github.com/user-attachments/assets/0e24ab3b-eece-45a2-95b6-3efe1357a6ba)

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

joshheald committed:
Track when a barcode scanning menu item is tapped

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  1 Message: <p>This PR contains changes to...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/247">22.7</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false"><p>This PR contains changes to Tracks-related logic. Please ensure (<strong>author and reviewer</strong>) the following are completed:</p>

<ul>
  <li>The tracks events must be validated in the Tracks system.</li>
  <li>Verify the internal Tracks spreadsheet has also been updated.</li>
  <li>Please consider registering any new events.</li>
  <li>The PR must be assigned the <strong>category: tracks</strong> label.</li>
</ul></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

joshheald commented:
https://github.com/Automattic/tracks-events-registration/pull/2994

joshheald committed:
Track barcode scanning dialog presentation

This is future proofing – in i2, this will likely be further down the flow than the primary tap on the menu item.

joshheald commented:
@staskus thanks for the review. FYI I've also added `barcode_scanning_explanation_dialog_shown` after discussion on other i2 analytics – it's tracked when the info screen is actually shown, as that will likely be moved further down the flow in a future update. Context: pdfdoF-7wE-p2#comment-8619

![CleanShot 2025-06-26 at 16 04 29@2x](https://github.com/user-attachments/assets/4ab7a077-c3c2-4cd2-86ee-0d08188e97f4)

I'll register and validate it when it shows up in Tracks.

joshheald commented:
https://github.com/Automattic/tracks-events-registration/pull/2997
